src/com/seafile/seadroid2/SSLTrustManager.java

```diff
@@ -239,14 +263,33 @@ private void customCheck(List<X509Certificate> chain, String authType)
                 reason = SslFailureReason.CERT_NOT_TRUSTED;
                 throw new CertificateException();
             } else if (savedCert.equals(cert)) {
-                // The user has confirmed to trust this certificate
-                Log.d(DEBUG_TAG, "the cert of " + account.server + " is trusted");
+                X509HostnameVerifier mHostnameVerifier = new BrowserCompatHostnameVerifier();
+                boolean verified = true;
+                if (mHostnameVerifier != null) {
+                    try {
+                        mHostnameVerifier.verify(account.server, cert);
+                    } catch (SSLException e) {
+                        verified = false;
+                    }
+
+                    if (!verified) {
+                        // check if user has trusted the host
+                        if (CertsManager.instance().getCertificate(account) != null) {
+                            Log.d(DEBUG_TAG, "the cert of " + account.server + " is trusted");
+                        } else {
+                            reason = SslFailureReason.CERT_NOT_TRUSTED;
+                            throw new CertificateException();
+                        }
+                    } else {
+                        Log.d(DEBUG_TAG, "the cert of " + account.server + " is trusted");
+                    }
+                }
```

Comment:
The above ~20 lines of code should be removed, since the hostname check part is moved to `validateHostName` method.